### ISSUE TITLE
Syncing stops randomly and without notification, when opening app it requires a new login without explaning why

### ISSUE THREAD

**Issue by henning** [nextcloud/ios#1497]
### Expected behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

App keeps syncing in background, when I open it, I see my data

### Actual behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               

After not having opened it for a while (a few weeks, maybe 3, maybe 6), it does not show me access to my data but presents a login form.

### Steps to reproduce     

Dont know hos to reproduce, because I dont know if the login data has been lost in some update or if the time of not opening the app directly is the trigger. Fact is my login credentials have not changed(other clients work) but login doesn't work anymore.

### Reasoning or why should it be changed/implemented?

The absolute main task of the app is reliably saving my data in the cloud, without permanent manual interaction.

Silently dropping login data and stopping sync without giving the users a chance to do something about it makes them vulnerable to major data loss.

### iOS version

Lates

### App version

Latest

### Server configuration

Is not under my control and to my best knowledge not related to the app's login behaviour.

If it would be the servers fault by changing login credentials or anything else, I would expect the app to notifiy me that it has problems to connect to the server, but that did not happen.


**Operating system:**

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)

**Comment by johkoenig** [comment]
Hello @henning,
can you please specify your app and iOS version? "Latest" isn't very precise.
Also, two notes concerning your problem:
1. Even if you change your password, you don't need to login again in the app, since the app is authorized based on a token. This also means that changing your password is not sufficient to deny access to devices that are not under your control anymore. 
2. You can check the logins and tokens to your nextcloud in the settings. See "Settings" -> "Security" -> "Devices & Sessions". Here, you can see which devices are connected to your account and when they have accessed your data used the last time.
Concerning the background syncing: This is not done automatically for all files, but only for those which are specifically tagged (e.g. starred files, or "keep always on device").

**Comment by henning** [comment]
@johkoenig thanks for getting back at me.

The version currently installed, since one week, is  3.0.15.
I also checked the server, it's on Nextcloud 20.0.1

Sorry for being too short-worded to post the exact version - but I also do not agree it is not sufficiently precise with respect to the properties of the problem!
 
The nature of the described issue implies, that it is not necessarily the currently installed version that made the problem occur first, so anyway to know more exactly which versions could be the source of the problem, somebody with access to the Appstore management must look which versions where provided to users at trhe time when the syncing supposedly has stopped. From the photos successfully uploaded I can tell, the last one was at  2020-09-18 - so the version with the bug must have been installed (likely automatically) on my phone on that day or the following, because usually no day passes without taking at least one picture of screenshot. To be safe, one could extend the time a bit and assume it must be the version available in the appstore for updates on existing installations, in Germany(in case that matters, I dont know how the appstore works exactly from a software distributors perspective), between 2020-09-10 and 2020-09-25. 

about 1)  I only stated that I did *not* change the login password to help exclude this as the source of the problem. 

about 2) 
I am not sure if you mean the iOS App or the web application. 

if you mean the iOS app:  I did describe the issue, and said, the app only displays a login screen (and I did not yet enter login data, because I didnt have the pasword at hand and were busy with other things), so there is no menu, security settings or whatnot to look at, just a login form. But  I will login again and look into it later.

The web app also does not offer any of the mentioned things. When I go to "settings" -> "Security" I see a page that says "Two-Factor Authentication", "server side encryption" and so forth, but none of the things you mention.

The other point:

Here it is true that my description was not precise. To clarify: With background syncing I do not mean "getting files from the server to the device automatically" but "getting my Photos synchronized to the server automatically" as it used to work for some time. (since the lat time a sync problem occured as reported in #1365) 

With the time of the issue having occured being quite a but in the past by now, and the situation being totally not reproducible, I believe it is not possible or worth to further trying to solve it. It might be interesting to keep an eye on it and take note if such a situation is reported more often, but until then nothing can be done with an acceptable amount of time and development energy.

I think it can be closed.
